### PR TITLE
Add x-exosite-token for service calling dispatcher (Issue: MUR-9171)

### PR DESCRIPTION
**Pull request by lenaw**

### CODE REVIEW

**Comment by lenaw** [comment]
@sauvainr , updated the doc to guide user how to use `x-exosite-token`, tell me if i missed something.
FYI @csrl

File: README.md [hunk 1]
```diff
@@ -744,3 +745,64 @@ paths:
 The definitions section can be used by reference objects in order re-use specification code. If elements of the spec are copied and pasted more than once, they can be set here and reused.
 
 ----
+
+#### Define a Token to Access Dispatcher
+We allow services, which are published on Exosite Marketplace, to access [dispatcher APIs](https://pegasus-dispatcher.hosted.exosite.io).
```
Comment (sauvainr) [inline README.md]:
```suggestion
We allow services, which are published on Exosite Marketplace, to access [dispatcher APIs](https://pegasus-dispatcher.hosted.exosite.io) using a security Token as defined in below steps.
```

File: README.md [hunk 2]
```diff
@@ -744,3 +745,64 @@ paths:
...
+
+##### Step 1: define `x-exosite-token`
+In `schema`, add a attribute called `x-exosite-token`, the value can be any string,
```
Comment (sauvainr) [inline README.md]:
```suggestion
In `schema`, add a attribute called `x-exosite-token`, the value needs to be a **secure token**  provided when accessing the Murano Service API.
```

File: README.md [hunk 3]
```diff
@@ -744,3 +745,64 @@ paths:
...
+it will be used as the token when you try to call dispatcher.
```
Comment (sauvainr) [inline README.md]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -744,3 +745,64 @@ paths:
...
+# ...
+################################################################################
+#                            API Information                                   #
+################################################################################
+# ...
+x-exosite-token: "myprivatetoken"
+# ...
+```
+
+So, we defined a service called `minimalservice` and the token is `myprivatetoken`.
```
Comment (sauvainr) [inline README.md]:
```suggestion
Once the service swagger has been published and processed, the token will be removed and will not be accessible from Murano anymore. 
So you MUST keep it securely saved. If you forget it you can overload the current token by updating the service swagger definition.
```

File: README.md [hunk 5]
```diff
@@ -744,3 +745,64 @@ paths:
...
+
+##### Step 2: call dispatcher
+
+Add a header `authorization: minimalservice myprivatetoken` for all the following endpoints.
```
Comment (sauvainr) [inline README.md]:
```suggestion
Add the `authorization` header with value `<service alias> <token value>` (Example:  `authorization: minimalservice myprivatetoken`) for all calls to the Murano Service API endpoints.
```

File: README.md [hunk 6]
```diff
@@ -744,3 +745,64 @@ paths:
...
+
+- 2.1 call Event Trigger
+
+  base_url = https://pegasus-dispatcher.hosted.exosite.io
```
Comment (sauvainr) [inline README.md]:
```suggestion
  base_url = https://murano-service-api.hosted.exosite.io
```
@csrl  @johnlinvc  any suggestion on the public api domain?

File: README.md [hunk 7]
```diff
@@ -744,3 +745,64 @@ paths:
...
+  - service_alias: which service you want to call, eg: tsdb
+  - event_type: which event to call, eg: export
+
+- 2.2 call Service Logs / Metrics
+
+  `GET/POST <base_url>/ws/service/:service/log`
+
+  `GET/POST <base_url>/ws/service/:service/metric/:metric`
+
+  If you want to use metric, please remember to define the `x-exosite-usage-metrics` along with `x-exosite-token`, if you don't define this, dispatcher will not add metrics for you.
```
Comment (sauvainr) [inline README.md]:
```suggestion
  If you want to use metric, please remember to define the `x-exosite-usage-metrics` along with `x-exosite-token`, if you don't define this, your connection to the metric endpoint will get rejected.
```

File: README.md (hunk 6 above)
Comment (lenaw) [reply README.md]: the current domain name is defined like this: [murano_ec2/development](https://github.com/exosite/murano_ec2/commit/e65470459c3d4c8749c401010c7381e4b0500373#diff-7741f37c8b6173cd1b798b99bd0cb48fR211),  should we change it to `serviceapis`? @johnlinvc  need your suggestions!!!

Comment (sauvainr) [reply README.md]: Actually it would be better to be:
> murano-service-api.exosite.io

File: README.md [hunk 8]
```diff
@@ -743,3 +745,64 @@ paths:
...
+Once the service swagger has been published and processed, the token will be removed and will not be accessible from Murano anymore. 
+So you MUST keep it securely saved. If you forget it you can overload the current token by updating the service swagger definition.
+
+##### Step 2: call dispatcher
+
+Add the `authorization` header with value `<service alias> <token value>` (Example:  `authorization: minimalservice myprivatetoken`) for all calls to the Murano Service API endpoints.
+
+- 2.1 call Event Trigger
+
+  base_url = https://murano-service-api.hosted.exosite.io
```
Comment (lenaw) [inline README.md]:
```suggestion
  base_url = https://murano-service-api.exosite.io
```

File: README.md [hunk 9]
```diff
@@ -743,3 +745,64 @@ paths:
...
+Once the service swagger has been published and processed, the token will be removed and will not be accessible from Murano anymore. 
+So you MUST keep it securely saved. If you forget it you can overload the current token by updating the service swagger definition.
+
+##### Step 2: call dispatcher
+
+Add the `authorization` header with value `<service alias> <token value>` (Example:  `authorization: minimalservice myprivatetoken`) for all calls to the Murano Service API endpoints.
+
+- 2.1 call Event Trigger
+
+  base_url = https://murano-service-api.exosite.io
```
Comment (lenaw) [inline README.md]:
```suggestion
  base_url = https://service-api.hosted.exosite.io
```

**Comment by lenaw** [comment]
@sauvainr  the public route is ready now. Is there any other thing needs to be updated on this PR?

**Comment by sauvainr** [comment]
@lenaw  no all good please go on, merge & push to QA.

**Comment by lenaw** [comment]
> @lenaw no all good please go on, merge & push to QA.

Need your approve to merge!

**Pull request merged**